### PR TITLE
api: sanity check addresses (round 1)

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

File: middleware/apimiddleware.go [hunk 1]
```diff
@@ -229,13 +231,22 @@ func GetBlockHashCtx(r *http.Request) (string, error) {
...
-func GetAddressCtx(r *http.Request) string {
-	address, ok := r.Context().Value(CtxAddress).(string)
+func GetAddressCtx(r *http.Request, activeNetParams *chaincfg.Params) (string, error) {
+	addressStr, ok := r.Context().Value(CtxAddress).(string)
 	if !ok {
 		apiLog.Trace("address not set")
-		return ""
+		return "", fmt.Errorf("address not set")
+	}
+	address, err := dcrutil.DecodeAddress(addressStr)
```
Comment (chappjc) [inline middleware/apimiddleware.go]: Seems OK, but as you see in `getAddressInfo` it is run again.

Comment (dajohi) [reply middleware/apimiddleware.go]: yes, future diff will be returning a diff type, if we go that way.

File: middleware/apimiddleware.go [hunk 2]
```diff
@@ -229,13 +231,22 @@ func GetBlockHashCtx(r *http.Request) (string, error) {
...
+	if err != nil {
+		return "", fmt.Errorf("invalid address '%v': %v",
+			addressStr, err)
+	}
+	if !address.IsForNet(activeNetParams) {
+		return "", fmt.Errorf("%v is invalid for this network",
+			address)
 	}
-	return address
+	return address.String(), nil
```
Comment (chappjc) [inline middleware/apimiddleware.go]: might as well return `addressStr` since we have it.

Comment (dajohi) [reply middleware/apimiddleware.go]: addressStr it is.

**Pull request merged**